lib/AST/Module.cpp

```diff
@@ -511,6 +511,27 @@ void ModuleDecl::addFile(FileUnit &newFile) {
   clearLookupCache();
 }
 
+ArrayRef<SourceFile *>
+PrimarySourceFilesRequest::evaluate(Evaluator &evaluator,
+                                    ModuleDecl *mod) const {
+  assert(mod->isMainModule() && "Only the main module can have primaries");
+
+  SmallVector<SourceFile *, 8> primaries;
+  for (auto *file : mod->getFiles()) {
+    if (auto *SF = dyn_cast<SourceFile>(file)) {
+      if (SF->isPrimary())
+        primaries.push_back(SF);
```

Comment:
Nit: I prefer straight-line code in loops like this, even though it can be more verbose.

```
for (auto *file : mod->getFiles()) {
  auto *SF = dyn_cast<SourceFile>(file);
  if (!SF || !SF->isPrimary()) { 
    continue; 
  }
  primaries.push_back(SF);
}
```

Reply:
I guess it's subjective, but for two conditions I find nesting to be slightly more readable. I'm happy to change it though :)

Reply:
FWIW, I have a slight preference for avoiding the negative logic of the continue. Just letting Hamish know he’s not alone.

Reply:
If only we had `guard`.

Reply:
yup, or this: ```
let primaries = mod.getFiles()
  .compactMap {$0 as? SourceFIle}
  .filter {$0.isPrimary()}
```

Reply:
Given the tests passed, I'm going to land this as-is, I'm more than happy to tweak the formatting in a follow-up.